federation/pkg/dnsprovider/providers/azure/azuredns/zones.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azuredns
+
+import (
+	"github.com/Azure/azure-sdk-for-go/arm/dns"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/federation/pkg/dnsprovider"
+)
+
+// Compile time check for interface adherence
+var _ dnsprovider.Zones = Zones{}
+
+// Zones holds a reference to the implementation of the provider specific interface_
+// that allows abstraction of the Azure DNS API.
+// During CI test, interface_ holds the mock implementation in
+// azurednsapi
+type Zones struct {
+	impl *Interface
+}
+
+// List returns all zones managed by this federation provider
+func (zones Zones) List() ([]dnsprovider.Zone, error) {
+	svc := zones.impl.service
+
+	azZoneList, err := svc.ListZones()
```

Comment:
Does ListZones return an empty list or an error if there are no zones.  The rest of the stack expects an empty list.  Worth checking.

Reply:
Verified. API does create an empty list